napari/components/viewer_model.py

```diff
@@ -263,8 +125,13 @@ def _scene_shape(self):
         # Scale the camera to the contents in the scene
         min_shape, max_shape = self._calc_bbox()
         size = np.subtract(max_shape, min_shape)
-        size = [size[i] for i in self.dims.displayed]
-        corner = [min_shape[i] for i in self.dims.displayed]
+
+        if not hasattr(self, 'dims'):
```

Comment:
why do we need this new `hasattr(self, 'dims')`, isn't this added in the `post_init` which will happen before it is ever needed?

Reply:
those are gone now too.  I needed to fix some things about the evented dataclass that was causing events to be emitted during the init.

Reply:
nice, yeah great not to need those!!

>  I needed to fix some things about the evented dataclass that was causing events to be emitted during the init.

It really is huge to be doing these PRs now to try this stuff out before we've merged everything, it's great to see all these little quirks that come up during real world usage!